src/clauseallocator.h

```diff
@@ -67,8 +67,8 @@ class ClauseAllocator {
                 throw CMSat::TooLongClauseError();
             }
 
-            void* mem = allocEnough(ps.size());
-            Clause* real = new (mem) Clause(ps, conflictNum
+            void* mem = allocEnough(ps.size() + (isAtmost ? 1 : 0));
```

Comment:
SInce we add an optional +1 element of the end of the clause in case of a cardinality constraint:
> Consequently, the allocation of this array has to be changed. I modified ClauseAllocator::Clause_new() in this direction. Is this the right spot?

Reply:
That +1 element should not be optional. It should be added to all clauses -- just not used for the ones that are not "isAtmost". This hacking for 32b of space is very confusing and will lead to massive amount of debugging overhead. I am pretty sure it's already wrong, so please just add that 32b into all clauses, then all these hacks will not be needed..